### PR TITLE
Feature/epo 139

### PR DESCRIPTION
**Pull request by Dragos0000**

### CODE REVIEW

File: src/html-conventions-lib/connectors-with-same-name.xsl [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    <xsl:import href="utils-html-conventions.xsl"/>
+
+    <xd:doc>
+        <xd:desc>Applying the checkers to a group of connectors with same name</xd:desc>
+    </xd:doc>
+
+    <xsl:template name="connectorsWithSameName">
+        <xsl:variable name="root" select="root()"/>
+        <xsl:variable name="distinctNames" select="f:getDistinctConnectorsNames($root)"/>
+        <h1>Connectors with the same name</h1>
```
Comment (costezki) [inline src/html-conventions-lib/connectors-with-same-name.xsl]: `<h1>Properties with multiple usages</h1>`
Move this section down, after the classes/enumerations/datatypes, just before any connector section.

File: src/html-conventions-lib/connectors-with-same-name.xsl [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+
+
+    <xd:doc>
+        <xd:desc>Check the multiplicity values from a group of connectors with same name</xd:desc>
+        <xd:param name="connectorName"/>
+        <xd:param name="root"/>
+    </xd:doc>
+    <xsl:template name="multiplicityForConnectorsWithSameName">
```
Comment (costezki) [inline src/html-conventions-lib/connectors-with-same-name.xsl]: `checkMultiplicityOfConnectorsWithSameName`

File: src/html-conventions-lib/connectors-with-same-name.xsl [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        <xsl:variable name="multiplicityValues"
+            select="$connectorsWithSameName/*[role/@name = $connectorName]/type/@multiplicity"/>
+        <xsl:variable name="allConnectorsHaveMultiplicityValue"
+            select="fn:count($connectorsWithSameName) = fn:count($multiplicityValues)"/>
+        <xsl:sequence
+            select="
+                if (f:areStringsEqual($multiplicityValues) and $allConnectorsHaveMultiplicityValue) then
+                    ()
+                else
+                    f:generateHtmlWarning('The multiplicity value used is not equal for all the connectors with this name')"
```
Comment (costezki) [inline src/html-conventions-lib/connectors-with-same-name.xsl]: `When a property is reused in multiple contexts, the multiplicity is expected to be the same. Please check the nomenclature above for a summary.`

File: src/html-conventions-lib/connectors-with-same-name.xsl [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        />
+    </xsl:template>
+
+
+    <xd:doc>
+        <xd:desc>Check the definition values from a group of connectors with same name</xd:desc>
+        <xd:param name="connectorName"/>
+        <xd:param name="root"/>
+    </xd:doc>
+    <xsl:template name="definitionForConnectorsWithSameName">
```
Comment (costezki) [inline src/html-conventions-lib/connectors-with-same-name.xsl]: rename `checkDefinitionOfConnectorsWithSameName`

File: src/html-conventions-lib/connectors-with-same-name.xsl [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            select="fn:count($connectorsWithSameName) = fn:count($definitionValues)"/>
+        <xsl:sequence
+            select="
+                if (f:areStringsEqual($definitionValues) and fn:boolean($definitionValues and $allConnectorsHaveDefinition)) then
+                    f:generateHtmlInfo(fn:concat('All the connectors with this name have the same definition. ',
+                    'Here is the usage: ',
+                    fn:string-join($descriptionsWithAnnotations, ',')))
+                else
+                    if (fn:boolean($definitionValues)) then
+                        f:generateHtmlWarning(fn:concat('The definition for the connectors with this name is different. ',
```
Comment (costezki) [inline src/html-conventions-lib/connectors-with-same-name.xsl]: The message is as follows: 

When a property is reused in multiple contexts, the meaning given by the definition is expected to be the same. In this case, multiple definitions are found.
_The text of the definition 1 in italic_ 
_(the connection name 1 in brackets and new line and italic)_
_The text of the definition 1 in italic and on a new line_ 
_(the connection name 2 in brackets and new line and italic)_
... etc.

File: src/html-conventions-report.xsl [hunk 6]
```diff
@@ -69,6 +69,7 @@
                     </div>
                     <xsl:call-template name="abstract"/>
                     <xsl:call-template name="infoBox"/>
+                    <xsl:call-template name="connectorsWithSameName"/>
```
Comment (costezki) [inline src/html-conventions-report.xsl]: move that below before all the connectors

File: test/unitTests/test-html-conventions-lib/test-connectors-with-same-name.xspec [hunk 7]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<x:description xmlns:x="http://www.jenitennison.com/xslt/xspec"
+    xmlns:xmi="http://www.omg.org/spec/XMI/20131001"
+    xmlns:f="http://https://github.com/costezki/model2owl#"
+    stylesheet="file:../../../src/html-conventions-lib/connectors-with-same-name.xsl">
+
+<!--    <x:scenario label="Scenario for testing template with match 'connectors">
```
Comment (costezki) [inline test/unitTests/test-html-conventions-lib/test-connectors-with-same-name.xspec]: we do not need commented source code.

File: test/unitTests/test-html-conventions-lib/test-connectors-with-same-name.xspec [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            <x:param name="connectorName">epo:hasPostalAddress</x:param>
+            <x:param name="root" href="../../testData/ePO-CM-v2.0.1-2020-06-08.eap.xmi" select="/"/>
+        </x:call>
+        <x:expect label="expect to find a Description Details element" test="boolean(/dd)"/>
+        <x:expect label="expect to find a info" test="/dd/@class = 'filter infos'"/>
+        
+    </x:scenario>
+    
+    <x:scenario
+        label="definition for connectors with the same name - not all the connectors compared have a definition">
```
Comment (costezki) [inline test/unitTests/test-html-conventions-lib/test-connectors-with-same-name.xspec:49]: very well done! Testing and testing and testing again!

File: src/example-report.html [hunk 9]
```diff
@@ -56,7 +56,7 @@ <h1>Abstract</h1>
                   Generalisations</em>. Each major section lists model items that need correction of errors
```
Comment (costezki) [inline src/example-report.html:56]: 1. please move the example report into the /data folder

2. please sort each of the nomenclature lists alphabetically.

**Review by costezki — APPROVED** [review]
well done! 

please implement the changes before the merge.

File: src/html-conventions-lib/connectors-with-same-name.xsl (hunk 5 above)
Comment (Dragos0000) [reply src/html-conventions-lib/connectors-with-same-name.xsl]: You can't have HTML markup in a string, sequence, select statement or fn:concat

File: src/example-report.html (hunk 9 above)
Comment (Dragos0000) [reply src/example-report.html:56]: done

**Pull request merged**